### PR TITLE
Fix out-of-PSR-6 requirements

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Just my 2cts that would allow https://github.com/symfony/symfony/pull/17408 to pass and still follow PSR-6

### CODE REVIEW

File: src/CachePoolTest.php [hunk 1]
```diff
@@ -316,22 +316,6 @@ public function testSave()
...
-        if (isset($this->skippedTests[__FUNCTION__])) {
-            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
-
-            return;
-        }
-
-        $item   = $this->cache->getItem('key');
-        $return = $this->cache->save($item);
-
-        $this->assertFalse($return, 'When saving an object with no data, we should generate an error.');
```
Comment (nicolas-grekas) [inline src/CachePoolTest.php:330]: Discussion about this started here: https://github.com/symfony/symfony/pull/17408#issuecomment-172456817

File: src/CachePoolTest.php [hunk 2]
```diff
@@ -721,7 +708,6 @@ public function testIsHit()
         $item = $this->cache->getItem('key');
         $item->set('value');
         $this->cache->save($item);
-        $this->assertTrue($item->isHit());
```
Comment (nicolas-grekas) [inline src/CachePoolTest.php:724]: Not a PSR-6 requirement to me.

File: src/CachePoolTest.php [hunk 3]
```diff
@@ -738,7 +724,6 @@ public function testIsHitDeferred()
         $item = $this->cache->getItem('key');
         $item->set('value');
         $this->cache->saveDeferred($item);
-        $this->assertTrue($item->isHit());
```
Comment (nicolas-grekas) [inline src/CachePoolTest.php:741]: Not a PSR-6 requirement to me.

File: src/CachePoolTest.php [hunk 4]
```diff
@@ -394,6 +380,7 @@ public function testDeferredSaveWithoutCommit()
         $this->cache->saveDeferred($item);
 
         $this->cache = null;
+        gc_collect_cycles();
```
Comment (nicolas-grekas) [inline src/CachePoolTest.php:383]: Required on PHP 7 to trigger the destructor immediately.

Comment (Nyholm) [inline src/CachePoolTest.php:383]: Cool, thank you

**Comment by Nyholm** [comment]
Ping @aequasi, can you give your opinion on this?

File: src/CachePoolTest.php (hunk 1 above)
Comment (kynx) [inline src/CachePoolTest.php:330]: :+1: I'm currently skipping this test over on my zend cache PR

File: src/CachePoolTest.php (hunk 3 above)
Comment (kynx) [inline src/CachePoolTest.php:741]: There was some discussion about this at https://groups.google.com/forum/?hl=en-GB#!topic/php-fig/pxy_VYgm2sU

**Comment by cryptiklemur** [comment]
Thanks @nicolas-grekas!

File: src/CachePoolTest.php (hunk 4 above)
Comment (wimleers) [inline src/CachePoolTest.php:383]: Then why not document this? Documentation for tricky things like this does not belong on GitHub, but in the code.

Comment (Nyholm) [inline src/CachePoolTest.php:383]: You are correct @wimleers. Would you mind sending a PR?

**Pull request merged**